src/main/java/de/dennisguse/opentracks/data/CustomContentProvider.java

```diff
@@ -148,22 +147,38 @@ public int delete(@NonNull Uri url, String where, String[] selectionArgs) {
         }
 
         Log.w(TAG, "Deleting table " + table);
-        int count;
+        int totalChangesBefore = getTotalChanges();
         try {
             db.beginTransaction();
-            count = db.delete(table, where, selectionArgs);
+            int count = db.delete(table, where, selectionArgs);
+            Log.i(TAG, "Deleted " + count + " rows of table " + table);
             db.setTransactionSuccessful();
         } finally {
             db.endTransaction();
         }
         getContext().getContentResolver().notifyChange(url, null, false);
 
-        if (shouldVacuum) {
-            // If a potentially large amount of data was deleted, reclaim its space.
+        int totalChanges = getTotalChanges() - totalChangesBefore;
+        Log.i(TAG, "Deleted " + totalChanges + " total rows of database");
+        return totalChanges;
+    }
+
+    private int getTotalChanges() {
+        int totalCount;
+        try (Cursor cursor = db.rawQuery("SELECT total_changes()", null)) {
+            cursor.moveToNext();
+            totalCount = cursor.getInt(0);
+        }
+        return totalCount;
+    }
+
+    @Override
+    public Bundle call(String method, String arg, Bundle extras) {
+        if (method.equals("vacuum")) {
```

Comment:
At least the string should be a constant.

Reply:
Sure